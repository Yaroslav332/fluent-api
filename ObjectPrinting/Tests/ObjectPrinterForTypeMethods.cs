using System;
using System.Globalization;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace ObjectPrinting.Tests
{
    [UseApprovalSubdirectory("ObjectPrinterForTypeMethods")]
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class ObjectPrinterForTypeMethods
    {
        [Test]
        public void ExcludeType_Should_Exclude()
        {
            var operation = new Operation()
            {
                Operator1 = 0.1f, Operator2 = 0.2f, Operator3 = 0.3f, Operator4 = 0.4, Operator5 = 0.5, Operator6 = 0.6,
                OperationDate = new DateTime(1994, 5, 3)
            };
            var printer = ObjectPrinter.For<Operation>()
                .ExcludeType<float>();
            Approvals.Verify(printer.PrintToString(operation));
        }
        
        [Test]
        public void AlternateType_Should_Alternate()
        {
            var operation = new Operation()
            {
                Operator1 = 0.1f, Operator2 = 0.2f, Operator3 = 0.3f, Operator4 = 0.4, Operator5 = 0.5, Operator6 = 0.6,
                OperationDate = new DateTime(1994, 5, 3)
            };
            var printer = ObjectPrinter.For<Operation>()
                .SetMethodForType((double d) => d + " hehe");
            Approvals.Verify(printer.PrintToString(operation));
        }
    }
}