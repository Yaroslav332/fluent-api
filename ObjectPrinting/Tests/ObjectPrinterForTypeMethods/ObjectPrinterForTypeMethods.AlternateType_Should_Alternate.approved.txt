Operation
	Operator1 = 0,1
	Operator2 = 0,2
	Operator3 = 0,3
	Operator4 = 0,4 hehe
	Operator5 = 0,5 hehe
	Operator6 = 0,6 hehe
	OperationDate = 03.05.1994 0:00:00
